lib/game.rb

```diff
@@ -1,7 +1,91 @@
 class Game
+  attr_reader :game_data
+
   def initialize(game_data)
     @game_data = game_data
   end
 
-  
+  def highest_total_score
+    highest_score = @game_data.map do |row|
+      row[:away_goals].to_i + row[:home_goals].to_i
+    end
+    highest_score.max
+  end
+
+  def lowest_total_score
+    lowest_score = @game_data.map do |row|
+      row[:away_goals].to_i + row[:home_goals].to_i
+    end
+    lowest_score.min
+  end
+
+  #Work in Progress
+  def count_of_games_by_season
+    season_hash = {}
+    @game_data.each do |game|
+      game.each do |key, value|
+        
+      end
+    end
+  end
+
+  # GAME STATS: Percentage Category; Home wins, Away wins, Ties. EB
+  def percentage_home_wins
+    home_wins = 0
```

Comment:
I wonder if we could refactor this if time permits. I do like your perspective with the counters to keep track of movement. Let's see how this method unfolds.